packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -93,7 +93,6 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
    * Invoked when a component is receiving new props. This method is not called for the initial render.
    */
   public componentWillReceiveProps(newProps: ISpinButtonProps): void {
-    this._lastValidValue = this.state.value;
```

Comment:
The original code is not right. `this.state.value` is not always a valid value. Think about this:

1. I clicked a button to send an AJAX request to update the form.
2. During the request, I modify the `SpinButton` and a invalid value. Now, `this.state.value` is invalid.
3. Keep the focus inside SpinButton to avoid blur, the AJAX request returns and update form including the SpinButton component. At this moment, `componentWillReceiveProps` is called and we cannot assume `this.state.value` as `lastValidValue`.

Besides, at a higher level design, we may not need a derived state from props. Read this: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
While I agree with this part of the PR, it's a potential change in logic. This should be split into it's own PR in case anything goes sideways with this the rest of your change will not be affected

---
In reply to: [246306986](https://github.com/OfficeDev/office-ui-fabric-react/pull/7577#discussion_r246306986) [](ancestors = 246306986)

Reply:
OK. I will make another PR for this. My this PR will depend on that PR.

Reply:
Oh, wait! My change will depend on this change because a controlled component has a more chance to set `_lastValidValue` to a wrong data.

1. Type some invalid value in SpinButton. Before blur, `this.state.value` is that invalid value.
2. When blur, SpinButton validates/normalizes the value and triggers the `onValidValueUpdated` with valid value. The parent calls `setState` and pass back the valid value to SpinButton. At the same time, SpinButton `setState` the valid value.
3. SpinButton runs to `componentWillReceiveProps`. However, at this moment, `this.state.value` is still using that old invalid value in step 1. Assign the value to `_lastValidValue` brings the SpinButton to a bad state.

So, it is a better decision to keep this line of change inside this pull request.